cmsplugin_zinnia/cms_toolbar.py

```diff
@@ -9,25 +9,42 @@
 class ZinniaToolbar(CMSToolbar):
 
     def populate(self):
+        user = self.request.user
         zinnia_menu = self.toolbar.get_or_create_menu(
             'zinnia-menu', _('Zinnia'))
 
         url = reverse('admin:zinnia_entry_add')
-        zinnia_menu.add_sideframe_item(_('New entry'), url=url)
+        zinnia_menu.add_sideframe_item(
+            _('New entry'), url=url,
+            disabled=not user.has_perm('zinnia.add_entry'))
 
         url = reverse('admin:zinnia_category_add')
-        zinnia_menu.add_sideframe_item(_('New category'), url=url)
+        zinnia_menu.add_sideframe_item(
+            _('New category'), url=url,
+            disabled=not user.has_perm('zinnia.add_category'))
 
         zinnia_menu.add_break()
 
         url = reverse('admin:zinnia_entry_changelist')
-        zinnia_menu.add_sideframe_item(_('Entries list'), url=url)
+        zinnia_menu.add_sideframe_item(
+            _('Entries list'), url=url,
+            disabled=not user.has_perm('zinnia.change_entry'))
 
         url = reverse('admin:zinnia_category_changelist')
-        zinnia_menu.add_sideframe_item(_('Categories list'), url=url)
+        zinnia_menu.add_sideframe_item(
+            _('Categories list'), url=url,
+            disabled=not user.has_perm('zinnia.change_category'))
 
         url = reverse('admin:tagging_tag_changelist')
-        zinnia_menu.add_sideframe_item(_('Tags list'), url=url)
+        zinnia_menu.add_sideframe_item(
+            _('Tags list'), url=url,
+            disabled=not user.has_perm('tagging.change_tag'))
+
+        # remove complete menu if all items are disabled
+        for item in zinnia_menu.get_items():
+            if hasattr(item, 'disabled') and not item.disabled:
+                return
```

Comment:
@Fantomas42 I've solved this with a list comprehension, reverting the logic. The code is much clearer now, it's finally easy to understand what it does. (Thanks!) - This way you don't need to install the plugin and do manual testing.

(Though, really, I should have added a test for this, shouldn't I?)